unitTests/ui/OmniboxTest.ts

```diff
@@ -10,16 +10,19 @@ import { IFieldDescription } from '../../src/rest/FieldDescription';
 import { Suggestion } from '../../src/magicbox/SuggestionsManager';
 import { KEYBOARD, OmniboxEvents } from '../../src/Core';
 import { IQueryOptions } from '../../src/controllers/QueryController';
+import { OmniboxAnalytics } from '../../src/ui/Omnibox/OmniboxAnalytics';
 
 export function OmniboxTest() {
   describe('Omnibox', () => {
-    var test: Mock.IBasicComponentSetup<Omnibox>;
+    let test: Mock.IBasicComponentSetup<Omnibox>;
+    let testEnv: Mock.MockEnvironmentBuilder;
```

Comment:
Why is this variable needed vs. using `test.env`?

Reply:
Ok, so the class is in the searchinterface.
`test.cmp.searchInterface` is `undefined`

the omniboxAnalytics is instantiated in the constructor of the Omnibox so I needed to create my spy before creating the mock component.

The only way I know is using the environment so I created a builder, added a spy that will return an instance of omniboxAnalytics and pass it as a reference when mocking my component.

Is there another way to achieve my goal here?